NXDOC-2925: Release Notes for nuxeo-ai 5.0.0 (#2637)
nuxeo/doc.nuxeo.com-platform-spaces: src/nxdoc/nuxeo-add-ons/nuxeo-ai/nuxeo-ai-release-notes/nuxeo-ai-5.0.0.md

#### Flexible Search Backend Support

AI-powered search and asset counting are no longer tied to a specific search engine. You can now swap or upgrade your search backend without disrupting AI features, giving you greater flexibility in how you architect your content platform. <br/>

### Bug Fixes

#### Image Quality Analysis Now Works Reliably With Sightengine

Previously, image quality checks via Sightengine could fail with errors, preventing image analysis results from being returned. This has been resolved — image quality and safety analysis now completes successfully. <br/>

#### AI Enrichment No Longer Fails Unexpectedly During Retries

In rare cases, AI enrichment processes could crash during automatic retries, leaving documents un-enriched. Retry handling is now more robust, ensuring enrichment completes reliably even when transient errors occur. <br/>

#### Accurate Document Counts in Search and Asset Reports

Search results and asset count reports could sometimes show incorrect totals. Document counts are now accurate across all search and reporting operations. <br/>

#### Reliable Face and Celebrity Detection for All Images

> Face and celebrity detection could fail on images that lacked orientation metadata (e.g., screenshots or programmatically generated images). These images are now processed successfully. <br/>

Reviewer: ```suggestion
Face and celebrity detection could fail on images that lacked orientation metadata (for example, screenshots or programmatically generated images). These images are now processed successfully. <br/>
```

#### Improved Memory Management During Image Processing

Image resizing operations could consume excess memory over time. Resource handling has been improved to ensure stable performance during bulk image processing. <br/>

### Improvements

#### More Reliable and Efficient Content Enrichment

> AI enrichment pipelines are now more resilient to unexpected responses from cloud providers, reducing the chance of enrichment failures. PDF text extraction uses less memory, improving performance for large documents. Image resizing now preserves the requested output format (PNG, GIF, etc.) instead of defaulting to JPEG, ensuring your assets maintain the quality you expect. <br/>

Reviewer: ```suggestion
AI enrichment pipelines are now more resilient to unexpected responses from cloud providers, reducing the chance of enrichment failures. PDF text extraction uses less memory, improving performance for large documents. Image resizing now preserves the requested output format (for examplem PNG, GIF) instead of saving the changes in JPEG format by default, ensuring your assets maintain the quality you expect. <br/>
```

#### Richer Image Analysis From Google Cloud Vision

You now get more detailed results from GCP Vision-powered analysis:

> - **Face detection** returns full emotion likelihoods (joy, anger, sorrow, surprise, etc.), bounding boxes, and confidence scores.

Reviewer: ```suggestion
- **Face detection** returns full emotion likelihoods (joy, anger, sorrow, surprise, and so on), bounding boxes, and confidence scores.
```

#### Pluggable Search Backend

A new search adapter service has been introduced to decouple AI features from specific search engine implementations. This enables backend replacement or upgrades without impacting AI functionality. <br/>

#### Docker Support

A Dockerfile has been added to support containerized deployments and simplify CI/CD integration. <br/>

### Bug Fixes

#### Sightengine API Returning HTTP 400 Errors

An issue was encountered when calling the Sightengine `/1.0/check.json` endpoint, which returned HTTP 400 responses. The root cause was the `model` parameter using `face` instead of `faces`. The model name has been corrected across all configuration files and documentation.

Additionally, the `Content-Type` for multipart text fields was changed from `application/octet-stream` to `text/plain; charset=UTF-8` to prevent API rejection or misinterpretation of request parameters. <br/>

#### Enrichment Provider Crashes on Null Exception Messages

> Retry logic across multiple enrichment providers could crash when processing exceptions that had no message. A null-check has been added to the retry policy abort predicates to handle this gracefully. <br/>

Reviewer: ```suggestion
Retry logic across multiple enrichment providers could crash when processing exceptions that do not have a message. A null-check has been added to the retry policy abort predicates to handle this gracefully. <br/>
```

#### Incorrect Hit Counts in Asset Counting and Search Operations

The fallback path in the search adapter service was using `size()` instead of `totalSize()`, which returned inaccurate document counts. This has been corrected. <br/>

#### Face and Celebrity Detection Crashes

> Potential crashes could occur in face and celebrity detection when image orientation metadata was unavailable. Null handling has been added to prevent these failures. <br/>

Reviewer: ```suggestion
Null handling has been added to prevent potential crashes that could have occurred in face and celebrity detection when image orientation metadata was unavailable. <br/>
```

#### Resource Leak in Image Resizing

A resource leak existed in the image resizing fallback path. This has been fixed to properly release resources. <br/>

#### Test Dependencies Included in Production Builds

Test-scoped dependencies were not properly scoped in the Maven build configuration, causing test-only libraries to be included in production artifacts. Dependency scopes have been corrected. <br/>

### Improvements

#### More Resilient Enrichment Pipelines

> Image and document enrichment pipelines have been hardened against unexpected API responses with comprehensive null handling. PDF text extraction has been optimized for better memory efficiency, especially for large documents. Image resizing now respects the requested output format (PNG, GIF, etc.) instead of always producing JPEG. <br/>

Reviewer: ```suggestion
Image and document enrichment pipelines have been hardened against unexpected API responses with comprehensive null handling. PDF text extraction has been optimized for better memory efficiency, especially for large documents. Image resizing now respects the requested output format (such as PNG, GIF) instead of always producing JPEG. <br/>
```

#### Richer Image Metadata from GCP Vision

> Full emotion likelihoods (joy, anger, sorrow, surprise, etc.), bounding boxes, and confidence scores have been restored for face detection. Bounding box geometry and proper classification tags have been added for logo and landmark detection. Label detection output format has been corrected for compatibility with downstream classification workflows. <br/>

Reviewer: ```suggestion
Full emotion likelihoods (such as joy, anger, sorrow, surprise), bounding boxes, and confidence scores have been restored for face detection. Bounding box geometry and proper classification tags have been added for logo and landmark detection. Label detection output format has been corrected for compatibility with downstream classification workflows. <br/>
```
